src/kudu/consensus/raft_consensus.cc

```diff
@@ -2229,17 +2229,26 @@ Status RaftConsensus::StartConsensusOnlyRoundUnlocked(const ReplicateRefPtr& msg
   VLOG_WITH_PREFIX_UNLOCKED(1) << "Starting consensus round: "
                                << SecureShortDebugString(msg->get()->id());
   scoped_refptr<ConsensusRound> round(new ConsensusRound(this, msg));
-  StdStatusCallback client_cb = std::bind(&RaftConsensus::MarkDirtyOnSuccess,
-                                          this,
-                                          string("Replicated consensus-only round"),
-                                          &DoNothingStatusCB,
-                                          std::placeholders::_1);
-  round->SetConsensusReplicatedCallback(std::bind(
-      &RaftConsensus::NonTxRoundReplicationFinished,
-      this,
-      round.get(),
-      std::move(client_cb),
-      std::placeholders::_1));
+  RETURN_NOT_OK(round_handler_->StartConsensusOnlyRound(round));
+
+  // Using disable_noop_ mode as a proxy for special NORCB handling
+  // When in disable_noop_ mode, the SetConsensusReplicatedCallback
+  // will be enqueued in the MySQL plugin.
+  // In this case we should not enqueue NonTxRoundReplicationFinished
+  // below, as it also does unsupported things, e.g. enqueing a CommitMsg
+  if (!disable_noop_) {
+    StdStatusCallback client_cb = std::bind(&RaftConsensus::MarkDirtyOnSuccess,
```

Comment:
One other side effect seems to be crashing when status wasn't OK. I think we didn't want it anyway. Right?

Reply:
The system reacts gracefully when StartConsensusOnlyRound status is NOT-OK. It rejects the message and the leader tries again. The Crash which you are probably talking about is that we crash if we have successfully prepared the message but actually could not append to the log file. This could happen if the disk is full. I think we should not crash the server in this case, and we are doing that today. But I was hoping that TLA+ would expose this gap and we would then put a right fix for it. I think that kuduraft will not work provably correct if we relax this, but MySQL should be 

Tell me if this is the crash you were talking about?

Reply:
Sure. Sounds good. I was talking about the `CHECK` calls to verify status::OK in callbacks. If the code path is exercised, we will get a SIGABRT.

Reply:
let me talk to you offline. i don't se the problematic CHECK.